lib/Parse/Lexer.cpp

```diff
@@ -656,6 +663,13 @@ static bool isRightBound(const char *tokEnd, bool isLeftBound,
     // "^.y" to be a prefix op, not binary.
     return !isLeftBound;
 
+  case '/':
+    // The start of a comment counts as whitespace, so it's not right bound.
+    if (tokEnd[1] == '/' || tokEnd[1] == '*')
+      return false;
+    else
+      return true;
+
```

Comment:
I think this is just a poorly-worded comment on my part. `tokEnd[1]` is actually past the end of the token we're deciding the right-boundedness of, so this is the case of a comment immediately following an operator character. I've tried to make this more clear.